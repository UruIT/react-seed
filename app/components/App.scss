@import '../styles/box-shadow.scss';

.shadow {
	@include box-shadow(1px, 1px, 4px,#4c4a4a);
}

.container {
	width: 420px;
	padding: 10px;
	margin: 32px auto;
	text-align: center;
}

.button {
	border: 2px solid #000;
	background-color: #fbfb09;
	cursor: pointer;
	font-weight: 600;
	font-size: 2rem;
	width: 200px;
	margin: 2rem;
}
