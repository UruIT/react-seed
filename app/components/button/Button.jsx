import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.scss';

const log = e => console.warn(e.target.textContent);

const Button = ({ text, onClick }) => (
	<div className={styles.container} onClick={onClick} >
		{ text }
	</div>
);

Button.propTypes = {
	onClick: PropTypes.func,
	text: PropTypes.string
};
Button.defaultProps = {
	onClick: log,
	text: '...'
}

export default Button;
