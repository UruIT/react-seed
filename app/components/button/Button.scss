.container {
	border: 1px solid black;
}
