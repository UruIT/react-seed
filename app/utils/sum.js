const sum = (a, b) => a + b;

const double = (a) => a + a;

export {
	sum,
	double
};
