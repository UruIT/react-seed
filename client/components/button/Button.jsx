import React from 'react';
import PropTypes from 'prop-types';
import styles from './button.scss';

const show = e => alert(e.target.textContent);

const Button = ({ text, onClick }) => (
	<div className={styles.container} onClick={onClick}>
		{ text }
	</div>
);

Button.propTypes = {
	onClick: PropTypes.func,
	text: PropTypes.string
};
Button.defaultProps = {
	onClick: show,
	text: '...'
};

export default Button;
