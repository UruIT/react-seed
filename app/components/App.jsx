import React from 'react';
import MyButton from './MyButton';
import styles from './App.scss';
import { getJson } from '../utils/fetch';

export default class App extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			joke: ''
		};
	}
	componentDidMount() {
		getJson('https://api.chucknorris.io/jokes/random')
			.then(res => this.setState({
				joke: res.value
			}));
	}
	render() {
		return (
			<div>
				<button className={styles.button} onClick={this.test}>Test</button>
				<MyButton>{ this.state.joke }</MyButton>
			</div>
		);
	}

	test = () => {
		console.error('test')
	}
}
