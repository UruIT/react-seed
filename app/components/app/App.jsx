import React from 'react';
import MyButton from '../button/Button';
import { getJson } from '../../utils/fetch';
import Counter from '../counter/Counter';
import styles from './App.scss';

export default class App extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			joke: ''
		};
	}
	handleClick = () => {
		return getJson('https://api.chucknorris.io/jokes/random')
			.then(res => this.setState({
				joke: res.value
			}));
	}
	render() {
		let { container, shadow, button } = styles;
		let { joke } = this.state;

		return (
			<div className={`${container} ${shadow}`}>
				<Counter />
				<button className={`${button} ${shadow}`} onClick={this.handleClick}>Test</button>
				{ !!joke && <MyButton text={joke} /> }
			</div>
		);
	}

	test = () => {
		console.error('test')
	}
}
