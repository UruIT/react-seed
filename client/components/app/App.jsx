import React from 'react';
import PropTypes from 'prop-types';
import Button from '../button/Button';
import Counter from '../counter';
import styles from './App.scss';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { jokeFetchRequested } from './app.action';

const { container, shadow, button } = styles;

export function App({ joke, loading, error, fetchJoke }) {
	let text = (loading && 'loading...') || error || joke;

	return (
		<div className={`${container} ${shadow}`}>
			<Counter />
			<button className={`${button} ${shadow}`} onClick={fetchJoke}>
				<FormattedMessage id="test" defaultMessage="Test" />
			</button>
			<div>
				{ !!text && <Button text={text} /> }
			</div>
		</div>
	);
}

App.propTypes = {
	fetchJoke: PropTypes.func.isRequired,
	joke: PropTypes.string.isRequired,
	loading: PropTypes.bool.isRequired,
	error: PropTypes.string.isRequired
};

const mapStateToProps = ({ app }) => app;
const mapDispatchToProps = dispatch => ({
	fetchJoke: () => dispatch(jokeFetchRequested())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
