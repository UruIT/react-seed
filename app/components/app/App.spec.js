jest.mock('../../utils/fetch', () => ({
	getJson: jest.fn(
		url =>
			new Promise(
				(resolve, reject) =>
					!!url ? resolve({ value: 'Chuck Norris joke' }) : reject()
			)
	)
}));

import React from 'react';
import App from './App';
import Button from '../button/Button';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { getJson } from '../../utils/fetch';

describe('On click in button', () => {
	const app = shallow(<App />);
	app.find('button').simulate('click');

	it('should call ChuckNorris API', () => {
		expect(getJson).toBeCalledWith(
			'https://api.chucknorris.io/jokes/random'
		);
	});

	it('should update Button children', () => {
		return app.instance().handleClick().then(() => {
			expect(app.find(Button).props().text).toEqual('Chuck Norris joke');
		});
	});
});

test('<App/>', () => {
	const tree = renderer.create(<App />).toJSON();

	expect(tree).toMatchSnapshot();
});
