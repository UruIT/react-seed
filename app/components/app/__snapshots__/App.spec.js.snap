// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> 1`] = `
<div
  className="container shadow"
>
  <div
    className="container"
  >
    <div>
      0
    </div>
    <a
      onClick={[Function]}
    >
      âˆ’
    </a>
    <a
      onClick={[Function]}
    >
      +
    </a>
  </div>
  <button
    className="button shadow"
    onClick={[Function]}
  >
    Test
  </button>
</div>
`;
